### PR TITLE
executor, tests: move test cases from `executor` to `integrationtest`

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: ref #47076

Problem Summary: update `batch_point_get`, `point_get`, `join`, `analyze`, `distsql`, `index_lookup_join` test cases. Move 50 test cases.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: tests/integrationtest/t/executor/point_get.test [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+prepare s from 'select * from t1 where id=1';
+set @@tidb_enable_plan_replayer_capture=1;
+execute s;
+execute s;
+connect (conn1, localhost, root, , executor__point_get);
+connection conn1;
+update t1 set v=v+1;
+connection default;
+disconnect conn1;
+execute s;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/point_get.test:336]: set @@tidb_enable_plan_replayer_capture=default;

File: tests/integrationtest/t/executor/join.test [hunk 2]
```diff
@@ -496,3 +496,75 @@ CREATE VIEW v0(c0) AS SELECT NULL;
 SELECT /*+ HASH_JOIN(v0) */ * FROM v0 NATURAL LEFT JOIN t0;
 SELECT /*+ MERGE_JOIN(v0) */ * FROM v0 NATURAL LEFT JOIN t0;
 --disable_warnings
+# TestTiDBNAAJ
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/join.test:499]:
```suggestion

# TestTiDBNAAJ
```

File: tests/integrationtest/t/executor/index_lookup_join.test [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+drop table if exists t2;
+create table t1 (a enum('x','y'));
+create table t2 (a int, key(a));
+insert into t1 values('x');
+insert into t2 values(1);
+select /*+ inl_join(t2) */ count(*) from t1 join t2 on t1.a = t2.a;
+select @@last_sql_use_alloc;
+select /*+ inl_hash_join(t2) */ count(*) from t1 join t2 on t1.a = t2.a;
+select @@last_sql_use_alloc;
+# Issue24547
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/index_lookup_join.test:206]: how about
```suggestion
## Issue24547
```
?

Comment (Defined2014) [reply tests/integrationtest/t/executor/index_lookup_join.test]: Updated.

**Review by SeaRise — APPROVED** [review]

**Pull request merged**